src/js/components/CheckBox/CheckBox.js

```diff
@@ -30,50 +34,105 @@ class CheckBox extends Component {
       ...rest
     } = this.props;
 
-    const normalizedLabel = (typeof label === 'string' ? <div>{label}</div> : label);
-
     let hidden;
     if (disabled && checked) {
       hidden = <input name={name} type='hidden' value='true' />;
     }
 
     const Icon = theme.checkBox.icons.checked;
 
+    let borderColor = normalizeColor(
+      theme.checkBox.border.color[theme.dark ? 'dark' : 'light'],
+      theme
+    );
+    if (checked) {
+      borderColor = normalizeColor(
+        (
+          theme.checkBox.color || theme.global.control.color
+        )[theme.dark ? 'dark' : 'light'],
+        theme
+      );
+    }
+
     const visual = (toggle ? (
-      <StyledCheckBoxToggle focus={focus} theme={theme}>
+      <StyledCheckBoxToggle focus={focus} theme={theme} checked={checked}>
         <StyledCheckBoxKnob theme={theme} />
       </StyledCheckBoxToggle>
     ) : (
-      <StyledCheckBoxBox focus={focus} theme={theme}>
+      <StyledCheckBoxBox
+        as={Box}
+        align='center'
+        justify='center'
+        width={theme.checkBox.size}
+        height={theme.checkBox.size}
+        border={{
+          size: theme.checkBox.border.width,
```

Comment:
Can we relabel to `theme.checkBox.border.size`?

Reply:
we have been using `width` all over `base.js` for border. I hesitant to change that. see control for an example:

```
control: {
        border: {
          width: '1px',
          radius: '4px',
          color: {
            dark: css`${props => props.theme.global.colors['border-dark']}`,
            light: css`${props => props.theme.global.colors['border-light']}`,
          },
        },
```

radiobutton uses that as well..